test/xbgpu/test_engine.py

```diff
@@ -959,72 +1021,58 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
-
-        incomplete_accums_counters = []
-        for i, corrprod_output in enumerate(corrprod_outputs):
-            # Or assert if incomplete_accs_total == incomplete_accums_counter * len(xbengine._pipelines)
-            incomplete_accums_counter = 0
-            base_batch_index = batch_start_index
-            for j, corrprod_result in enumerate(corrprod_results[i]):
-                # The first heap is an incomplete accumulation containing a
-                # single batch, we need to make sure that this is taken into
-                # account by the verification function.
-                if j == 0:
-                    # This is to handle the first accumulation processed. The value
-                    # checked here is simply the first in the range.
-                    # - Even though :func:`generate_expected_corrprods` returns a
-                    #   zeroed array for an incomplete accumulation, we still need
-                    #   to maintain programmatic sense in the values generated here.
-                    num_batches_in_current_accumulation = HEAPS_PER_FENGINE_PER_CHUNK
-                    incomplete_accums_counter += 1
-                else:
-                    num_batches_in_current_accumulation = corrprod_output.heap_accumulation_threshold
-                    if missing_antenna is not None:
-                        incomplete_accums_counter += 1
-
-                expected_output = generate_expected_corrprods(
-                    base_batch_index,
-                    num_batches_in_current_accumulation,
-                    corrprod_output.heap_accumulation_threshold,
-                    n_channels_per_substream,
-                    n_ants,
-                    n_spectra_per_heap,
-                    missing_antenna,
+            with caplog.filtering(AccumWarningFilter()):
+                corrprod_results, beam_results = await self._send_data(
+                    mock_recv_streams,
+                    mock_send_stream,
+                    corrprod_outputs=corrprod_outputs,
+                    beam_outputs=beam_outputs,
+                    batch_indices=test_batch_indices,
+                    heap_factory=heap_factory,
+                    timestamp_step=timestamp_step,
+                    n_ants=n_ants,
+                    n_channels_per_substream=n_channels_per_substream,
+                    frequency=frequency,
+                    n_spectra_per_heap=n_spectra_per_heap,
+                    missing_antennas=missing_antennas,
                 )
-                base_batch_index += num_batches_in_current_accumulation
-                np.testing.assert_equal(expected_output, corrprod_result)
-            incomplete_accums_counters.append(incomplete_accums_counter)
+            last_timestamp = batch_end_index2 * timestamp_step
+
+        # TODO: NGC-1308 Update this check to not be a subset of the warnings filtered
+        assert caplog.record_tuples[: len(corrprod_outputs)] == [
```

Comment:
It looks like you've put in the filtering now, and could remove the `: len(corrprod_outputs)`?

Reply:
Unfortunately not. Some combinations of `heap_accumulation_thresholds` - `(3,7), (5, 9)` - still report the "All Antennas [...]" warning during that jump in data in the middle of the test (the artificial break in data we introduced).
* Those warnings may technically be true, but not exactly because of our decision to start 'in the middle of a Chunk' at the start of the test.

Let me know if that does/doesn't make sense.